src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/RoslynServices.cs

```diff
@@ -25,19 +26,11 @@ internal class RoslynServices : IRoslynServices
         [ImportMany]
         protected OrderPrecedenceImportCollection<ISyntaxFactsService> SyntaxFactsServicesImpl { get; }
 
-        private ISyntaxFactsService? SyntaxFactsService
-        {
-            get
-            {
-                return SyntaxFactsServicesImpl.FirstOrDefault()?.Value;
-            }
-        }
+        private ISyntaxFactsService? SyntaxFactsService => SyntaxFactsServicesImpl.FirstOrDefault()?.Value;
 
         public Task<Solution> RenameSymbolAsync(Solution solution, ISymbol symbol, string newName, CancellationToken token = default)
         {
-#pragma warning disable CS0618 // Type or member is obsolete https://github.com/dotnet/project-system/issues/8591
-            return RoslynRenamer.Renamer.RenameSymbolAsync(solution, symbol, newName, solution.Workspace.Options, token);
-#pragma warning restore CS0618 // Type or member is obsolete
+            return Renamer.RenameSymbolAsync(solution, symbol, s_renameOptions, newName, token);
```

Comment:
We never updated the options provided by `solution.Workspace.Options`. Behind the scenes, it is just a new'ed `SymbolRenameOptions` but there is a translation layer between that class (which is the newer options class) and the older one provided by the `Options` property (which is also deprecated).

Reply:
My reading of this is that it should pull settings from the solution. Something like:

```suggestion
            SymbolRenameOptions options = new(
                RenameOverloads: solution.Options.GetOption(RenameOptions.RenameOverloads),
                RenameInStrings: solution.Options.GetOption(RenameOptions.RenameInStrings),
                RenameInComments: solution.Options.GetOption(RenameOptions.RenameInComments),
                RenameFile: false);                

            return Renamer.RenameSymbolAsync(solution, symbol, options, newName, token);
```

`solution.Workspace.Options` ends up reading the `OptionSet` from the underlying solution, so I think the above would avoid a change of behaviour in this code.

We could consider extending our dialog to display these options too.